### PR TITLE
log refactor: dm-master

### PR DESCRIPTION
**Pull request by GregoryIan**
### What problem does this PR solve? 
refactor log of dmctl

### What is changed and how it works?
use pingcap/log

Pr recommended review order
https://github.com/pingcap/dm/pull/162
https://github.com/pingcap/dm/pull/170
https://github.com/pingcap/dm/pull/172
https://github.com/pingcap/dm/pull/173
https://github.com/pingcap/dm/pull/175
https://github.com/pingcap/dm/pull/176

### CODE REVIEW

**Comment by WangXiangUSTC** [comment]
this pr commit to branch `log`, but merge `master`

**Comment by GregoryIan** [comment]
@WangXiangUSTC any problem now?

File: dm/master/server.go [hunk 1]
```diff
@@ -134,12 +136,12 @@ func (s *Server) Start() error {
 	go func() {
 		err2 := s.svr.Serve(grpcL)
 		if err2 != nil && !common.IsErrNetClosing(err2) && err2 != cmux.ErrListenerClosed {
-			log.Errorf("[server] gRPC server return with error %s", err2.Error())
+			log.L().Error("fail to start gRPC server", zap.Error(err2))
```
Comment (WangXiangUSTC) [inline dm/master/server.go]: how about use ShortError

File: dm/master/server.go [hunk 2]
```diff
@@ -134,12 +136,12 @@ func (s *Server) Start() error {
...
 		}
 	}()
 	go InitStatus(httpL) // serve status
 
-	log.Infof("[server] listening on %v for gRPC API and status request", s.cfg.MasterAddr)
+	log.L().Info(" listening gRPC API and status request", zap.String("address", s.cfg.MasterAddr))
```
Comment (WangXiangUSTC) [inline dm/master/server.go]: how about define a variable `logger` in the `master`, and set logger with prefix `server` in `init` function?

File: dm/master/server.go [hunk 3]
```diff
@@ -159,7 +161,7 @@ func (s *Server) Close() {
 	}
 	err := s.rootLis.Close()
 	if err != nil && !common.IsErrNetClosing(err) {
-		log.Errorf("[server] close net listener with error %s", err.Error())
+		log.L().Error("close net listener", zap.Error(err))
```
Comment (WangXiangUSTC) [inline dm/master/server.go]: how about use `ShortError`?

File: dm/master/server.go [hunk 4]
```diff
@@ -956,7 +958,7 @@ func (s *Server) addTaskWorkers(task string, workers []string, replace bool) {
 
 	sort.Strings(valid)
 	s.taskWorkers[task] = valid
-	log.Infof("[server] update task %s workers to %v", task, valid)
+	log.L().Info("update workers of task", zap.String("task", task), zap.Reflect("workers", valid))
```
Comment (WangXiangUSTC) [inline dm/master/server.go]: how about use `zap.Strings` for `valid`

File: dm/master/server.go (hunk 2 above)
Comment (GregoryIan) [reply dm/master/server.go]: what's the definition of `server`?

Comment (WangXiangUSTC) [reply dm/master/server.go]: image

Comment (WangXiangUSTC) [reply dm/master/server.go]: like this in my pr https://github.com/pingcap/dm/pull/170

Comment (GregoryIan) [reply dm/master/server.go]: I choose to ignore `server` label  to indicate  `I'm main process`, but I prefer to add `component` for other independent component

File: dm/master/server.go (hunk 4 above)
Comment (GregoryIan) [reply dm/master/server.go]: changed

**Comment by WangXiangUSTC** [comment]
LGTM

File: dm/master/server.go [hunk 5]
```diff
@@ -1565,7 +1568,7 @@ func (s *Server) allWorkerConfigs(ctx context.Context) (map[string]config.DBConf
 	)
 	handErr := func(err2 error) {
 		if err2 != nil {
-			log.Error(err2)
+			log.L().Error(err2.Error())
```
Comment (amyangfei) [inline dm/master/server.go]:
```suggestion
			log.L().Error(zap.Error(err2))
```

File: dm/master/sql-operator/operator.go [hunk 6]
```diff
@@ -132,7 +135,7 @@ func (h *Holder) Get(taskName string, sqls []string) (string, *Operator) {
 	for _, sql := range sqls {
 		for key, oper := range operators {
 			if oper.matchPattern(sql) { // matched one SQL of all is enough
-				log.Infof("[sql-operator] get an operator %s with key %s matched SQL %s", oper, key, sql)
+				h.logger.Info("get an matched operator", zap.Stringer("operation", oper), zap.String("key", key), zap.String("sql", sql))
```
Comment (amyangfei) [inline dm/master/sql-operator/operator.go]:
```suggestion
				h.logger.Info("get a matched operator", zap.Stringer("operation", oper), zap.String("key", key), zap.String("sql", sql))
```

File: dm/master/server.go [hunk 7]
```diff
@@ -23,6 +23,8 @@ import (
 	"sync"
 	"time"
 
+	"go.uber.org/zap"
```
Comment (amyangfei) [inline dm/master/server.go]: it is better to re-organize import group based on DM custom regulation. ditto in some other files.

Comment (GregoryIan) [reply dm/master/server.go]: I would re-organize all import path after we refactor log

**Comment by GregoryIan** [comment]
PTAL again @amyangfei

**Pull request merged**